Tasks/PackerBuildV0/src/operations/packerBuild.ts

```diff
@@ -26,23 +26,44 @@ export async function run(packerHost: packerHost): Promise<any> {
     command.arg(packerHost.getTemplateFileProvider().getTemplateFileLocation(packerHost));
 
     console.log(tl.loc("ExecutingPackerBuild"));
-    var outputVariablesParser: definitions.IOutputParser = new OutputVariablesParser([constants.PackerLogTokenImageUri, constants.PackerLogTokenStorageLocation]);
+    var taskParameters = packerHost.getTaskParameters();
+    var outputVariablesParser: definitions.IOutputParser;
+    if(!taskParameters.isManagedImage){
+        outputVariablesParser = new OutputVariablesParser([constants.PackerLogTokenImageUri, constants.PackerLogTokenStorageLocation]);
+    } else {
+        outputVariablesParser = new OutputVariablesParser([constants.PackerLogTokenManagedImageName, constants.PackerLogTokenManagedResourceGroupName, constants.PackerLogTokenManagedImageLocation]);
+    }
+
     await packerHost.execTool(command, outputVariablesParser);
 
     // set output task variables
     setOutputVariables(packerHost, outputVariablesParser.getExtractedOutputs());
 }
 
 function setOutputVariables(packerHost: packerHost, outputs: Map<string, string>): void {
-    var imageUri = outputs.get(constants.PackerLogTokenImageUri);
-    var taskParameters = packerHost.getTaskParameters();
+   var taskParameters = packerHost.getTaskParameters();
+   var imageUri;
+   if(!taskParameters.isManagedImage) {
+       imageUri = outputs.get(constants.PackerLogTokenImageUri);
+   } else {
+       imageUri = outputs.get(constants.PackerLogTokenManagedImageName);
+   }
 
     if(!utils.IsNullOrEmpty(taskParameters.imageUri)) {
         if(!utils.IsNullOrEmpty(imageUri)) {
-            tl.debug("Setting image URI variable to: " + imageUri);
+            if(!taskParameters.isManagedImage){
+                tl.debug("Setting image URI variable to: " + imageUri);
+            } else {
+                tl.debug("Setting image URI variable which contains the managed image name to: " + imageUri);
+            }
+            
             tl.setVariable(taskParameters.imageUri, imageUri);
```

Comment:
Should we have a separate output variable for managed image name? we can show output variable based on whether isManagedImage is true. @azooinmyluggage any thoughts?

Reply:
Discussed with @azooinmyluggage. This is fine the way it is.